### PR TITLE
Allow publishing to Hex

### PR DESCRIPTION
**Pull request by estreeper**
This _should_ deploy a new version to Hex when we create a release via GitHub, and also introduces testing back to Elixir version 1.12/OTP24.

Also includes a script to run the CI checks locally, and instructions on how to make it run automatically pre-push.

Minor documentation updates to specify we want PRs linked to an Issue.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -35,25 +38,14 @@ jobs:
           restore-keys: |
             ${{ runner.os }}-mix-${{ env.cache-name }}-
 
-      - name: Cache compiled build
```
Comment (estreeper) [inline .github/workflows/test.yml:38]: This seemed to be causing a BEAM version mismatch, I suppose because of the two versions (1.12/OTP24 and 1.16/OTP26) being tested. It didn't seem to appreciably increase CI run time (it's still <50s)

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -42,10 +45,10 @@ jobs:
           cache-name: cache-compiled-build
         with:
           path: _build
-          key: ${{ runner.os }}-mix-${{ env.cache-name }}-${{ hashFiles('**/mix.lock') }}
+          key: ${{ runner.os }}-mix-${{ env.cache-name }}-elixir-${{ matrix.elixir }}-${{ hashFiles('**/mix.lock') }}
           restore-keys: |
-            ${{ runner.os }}-mix-${{ env.cache-name }}-
-            ${{ runner.os }}-mix-
```
Comment (estreeper) [inline .github/workflows/test.yml:48]: I had to make these changes because otherwise I believe it was trying to use the same cached version for both combos (1.12/OTP24, 1.16/OTP26) and throwing an error about a BEAM version mismatch

File: test/prodops_ex/artifact_test.exs [hunk 3]
```diff
@@ -93,7 +93,7 @@ defmodule ProdopsEx.ArtifactTest do
   describe "stream_refine_artifact/2" do
     test "streams refinement of an artifact by submitting a request with the required parameters" do
       config = [api_url: "https://api.example.com", api_key: "secret_key"]
-      params = %{artifact_id: 1, artifact_slug: "story"}
```
Comment (estreeper) [inline test/prodops_ex/artifact_test.exs:96]: This was a driveby fix: this test had broken during a merge

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -54,6 +57,7 @@ jobs:
         run: mix compile --warnings-as-errors
 
       - name: Check formatting
+        if: matrix.elixir == '1.16' && matrix.otp == '26'
```
Comment (estreeper) [inline .github/workflows/test.yml:60]: The old formatter has different opinions, so just used the newer one

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -67,6 +71,7 @@ jobs:
             plt-cache-
 
       - name: Check function specs with Dialyzer
+        if: matrix.elixir == '1.16' && matrix.otp == '26'
```
Comment (estreeper) [inline .github/workflows/test.yml:74]: This wasn't causing a problem but also figured this could be skipped since the new dialyzer is probably better

File: .github/workflows/publish-hex.yml [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+      - name: Check function specs with Dialyzer
+        if: matrix.elixir == '1.16' && matrix.otp == '26'
+        run: mix dialyzer
+
+      - name: Run tests
+        run: mix test
+
+      - name: Publish to Hex
+        run: mix hex.publish --yes
```
Comment (grossvogel) [inline .github/workflows/publish-hex.yml:51]: It seems to me like this will cause `mix hex.publish` to happen twice, once for each entry in the matrix. IMO we can probably just remove 1.12 from the matrix since every commit should have passed the other workflow before we tag it as a release

File: .github/workflows/publish-hex.yml [hunk 7]
```diff
@@ -0,0 +1,51 @@
+name: Publish to Hex
+
+on:
+  release:
+    types: [published]
+
+jobs:
+  publish:
+    runs-on: ubuntu-22.04
+    name: Test on OTP ${{matrix.otp}} / Elixir ${{matrix.elixir}}
```
Comment (grossvogel) [inline .github/workflows/publish-hex.yml:10]:
```suggestion
    name: Publish to Hex
```

**Review by grossvogel — COMMENTED** [review]
I think we can just use the latest supported Elixir version for the publish workflow, otherwise everything looks good

File: .github/workflows/test.yml (hunk 5 above)
Comment (grossvogel) [reply .github/workflows/test.yml:74]: yeah nobody wants to wait for dialyzer to run twice
